tests/testsuite.at

```diff
@@ -203,6 +203,49 @@ m4_define([NEGATIVE_TEST],[
   ])
 ])
 
+m4_define([SWIG_LANGUAGES], [[java],
+  [python]])
+
+dnl Positive testcase for SWIG given configuration
+dnl $1 test name
+dnl $2 category
+dnl $3 language
+m4_define([TEST_EVAL_SWIG],[
+  m4_define([TESTNAME],[$1])
+  m4_define([CATEGORY],[$2])
+  m4_define([LANGUAGE],[$3])
+  m4_define([SCRIPTDIR],["$TESTS"/CATEGORY/LANGUAGE])
+  m4_define([TESTDIR],[SCRIPTDIR/TESTNAME])
+  m4_define([PROGRAM],[TESTDIR/TESTNAME.dl])
+  m4_define([FACTS],[TESTDIR/facts])
+  m4_define([EXPECTEDDIR],[TESTDIR/expected])
+
+  # clean files before testing
+  m4_esyscmd_s(.SCRIPTDIR/clean-test-files-LANGUAGE.sh CATEGORY/LANGUAGE/TESTNAME)
+
+  AT_CHECK("$SOUFFLE" -D. -F FACTS PROGRAM)
```

Comment:
Why not use `-D EXPECTEDDIR`? This way you don't need to make the directory and move stuff there (if it already exists, otherwise you will have to `mkdir EXPECTEDDIR` first).